plugins/async-node/jvm/src/test/kotlin/com/intuit/playerui/plugins/asyncnode/AsyncNodePluginTest.kt

```diff
@@ -228,4 +228,64 @@ internal class AsyncNodePluginTest : PlayerTest() {
         }
         Assertions.assertTrue(true)
     }
+
+    @TestTemplate
+    fun `handle multiple updates through callback mechanism`() = runBlockingTest {
+        var update: Asset? = null
+        var updateCount = 0
+
+        plugin?.hooks?.onAsyncNode?.tap("") { _, node, callback ->
+            when (updateCount) {
+                0 -> {
+                    callback(mapOf("asset" to mapOf("id" to "asset-1", "type" to "text", "value" to "First update")))
+                }
+
+                1 -> {
+                    callback(mapOf("asset" to mapOf("id" to "asset-2", "type" to "text", "value" to "Second update")))
+                }
+
+                2 -> {
+                    callback(mapOf("asset" to mapOf("id" to "asset-3", "type" to "text", "value" to "Third update")))
+                }
+            }
+            updateCount++
+            BailResult.Bail(emptyList())
+        }
+
+        var count = 0
+        suspendCancellableCoroutine { cont ->
+            player.hooks.view.tap { v ->
+                v?.hooks?.onUpdate?.tap { asset ->
+                    count++
+                    update = asset
+                    if (count == 3) cont.resume(true) {}
+                }
+            }
+            player.start(asyncNodeFlowSimple)
+        }
+
+        Assertions.assertTrue(count == 3)
+        Assertions.assertTrue((update?.get("actions") as List<*>).isNotEmpty())
+    }
+
+    @TestTemplate
+    fun `handle undefined node`() = runBlockingTest {
+        var update: Asset? = null
+        plugin?.hooks?.onAsyncNode?.tap("") { _, node, callback ->
+            BailResult.Bail(emptyList())
+        }
+        var count = 0
+        suspendCancellableCoroutine { cont ->
+            player.hooks.view.tap { v ->
+                v?.hooks?.onUpdate?.tap { asset ->
+                    count++
+                    update = asset
+                    if (count == 2) cont.resume(true) {}
+                }
+            }
+            player.start(asyncNodeFlowSimple)
+        }
+        Assertions.assertTrue(count == 2)
+        Assertions.assertTrue((update?.get("actions") as List<*>).isNotEmpty())
```

Comment:
similar comment here, but in this case, is an `emptyList` representative of `undefined`? and if so, should this be not empty?

Reply:
I just realized this brings up another question, should this get a second update at all if you're resolving it with "`undefined`" right away?